### DISCUSSION TITLE
Extracting amp/freq pairs from fluid.sines~

### DISCUSSION THREAD

**Discussion by jamesb93** [flucoma/flucoma-core#131] [Ideas]
I think we've heard this one before too. People like the idea of being able to know the frequency/amplitude pairs of the sinusoidal component of `fluid.sines~`. Again, I'm approaching this from a pie in the sky viewpoint and I don't understand if it is really a component of the algorithm under the hood but it might be Another Useful Thing to play with and perhaps function a bit like `sigmund~` does.

@weefuzzy, because you are the resident C++ warlock these days and I know all feature requests basically make work for you what do you think of this?

**Comment by tremblap** [comment]
There is an old ticket with this (like 4 years old) in bitbucket... I am likely to use that or something similar as my first client coding in the summer, so keep on putting ideas here but I'd say the chances of having any new features before mid-July (aka v1) are infinitesimal. Still very good to have the ideas here so keep them coming.

**Comment by weefuzzy** [comment]
I guess it's similar work to `NoveltyFeature` et al: decoupling the algorithm and producing a new object. Not a huge amount on the surface, but there are bound to be edges. The first edge is whether we want picked peaks (like `sigmund~`'s default mode) or the tracked partials as output. (I know the answer will be 'both', but this is to get the ball rolling on someone else thinking about the interface repercussions of that...)
